vms/platformvm/vm.go

```diff
@@ -205,7 +205,10 @@ func (vm *VM) Initialize(
 		chainCtx.Log,
 		chainCtx.NodeID,
 		chainCtx.SubnetID,
-		chainCtx.ValidatorState,
+		validators.NewLockedState(
+			&chainCtx.Lock,
+			validatorManager,
+		),
```

Comment:
This is the actual fix. `chainCtx.ValidatorState` was replaced with `validatorManager` because `chainCtx.ValidatorState` is actually equal to `vm` here and I found that _very_ confusing. Additionally, this made testing much better because we don't need to rely on mocking the `ValidatorState`.